common/tools/dev-tool/src/commands/samples/publish.ts

```diff
@@ -0,0 +1,615 @@
+
+  const defaultDependencies: Record<string, string> = {
+    // If we are a beta package, use "next", otherwise we will use "latest"
+    [projectInfo.name]: projectInfo.version.includes("beta") ? "next" : "latest",
+    // We use this universally
+    dotenv: "latest"
+  };
+
+  const { packageJson } = projectInfo;
+
+  if (!sampleConfiguration.productSlugs && !sampleConfiguration.disableDocsMs) {
+    log.error("No extra product slugs provided (`productSlugs` in the sample configuration).");
+    log.warn(
+      "There is probably a more specific product that applies to this package! Reach out for help with the docs platform."
+    );
+    log.warn(
+      'If you do not want to publish samples to docs.microsoft.com, set `"disableDocsMs": true` in the sample configuration.'
+    );
+    onError();
+  }
+
+  return {
+    ...sampleConfiguration,
+    baseName,
+    packageKeywords:
+      projectInfo.packageJson.keywords ??
+      fail(`The package.json for ${projectInfo.name} does not specify "keywords".`),
+    projectRepoPath:
+      "sdk/" +
+      projectInfo.path
+        .split(path.sep + "sdk" + path.sep)
+        .slice(-1)[0]
+        .replace("\\", "/"),
+    // This'll be good enough most of the time, but products like Azure Form Recognizer will have
+    // too adjust using the sample configuration.
+    productName:
+      sampleConfiguration.productName ??
+      fail(`The sample configuration does not specify a "productName".`),
+    apiRefLink: sampleConfiguration.apiRefLink,
+    sampleSourcesPath,
+    topLevelDirectory,
+    moduleInfos,
+    computeSampleDependencies(outputKind: OutputKind) {
+      return {
+        dependencies: moduleInfos.reduce((prev, source) => {
+          const current: Record<string, string> = {};
+          for (const dependency of source.importedModules) {
+            if (prev[dependency] === undefined) {
+              const dependencyVersion =
+                sampleConfiguration.dependencyOverrides?.[dependency] ??
+                packageJson.devDependencies[dependency] ??
+                packageJson.dependencies[dependency];
+              if (dependencyVersion === undefined) {
+                log.error(
+                  `Dependency "${dependency}", imported by ${source.filePath}, has an unknown version.`
+                );
+                log.error(
+                  "Specify a version for it by including it in the package's `devDependencies`."
+                );
+              }
+
+              current[dependency] = dependencyVersion;
+            }
+          }
+          return {
+            ...prev,
+            ...current
+          };
+        }, defaultDependencies),
+        ...(outputKind === OutputKind.TypeScript
+          ? {
+              // In TypeScript samples, we include TypeScript and `rimraf`, because they're used
+              // in the package scripts.
+              devDependencies: {
+                typescript: "~4.1.0",
```

Comment:
wonder if TS version should get pulled out to a variable somewhere so we don't forget to update it... or if you wanna be really fancy grab this from the package.json of dev-tool itself... :)

Reply:
> if you wanna be really fancy grab this from the package.json of dev-tool itself

Wondering why I didn't do this in the first place.

Reply:
I made it take the version from dev-tool's dependencies.